src/app.py

```diff
@@ -71,7 +71,20 @@ async def current_user(authorization: Optional[str] = Header(None)) -> User:
     if bearer != "Bearer":
         return User(None)
 
-    return User(token)
+    secret = db.get_mquery_config_key("openid_secret")
+    if secret is None:
+        return User(None)
+
+    public_key = serialization.load_der_public_key(base64.b64decode(secret))
+    try:
+        token_json = jwt.decode(
+            token, public_key, algorithms=["RS256"], audience="account"  # type: ignore
+        )
+    except jwt.ExpiredSignatureError:
```

Comment:
What will happend when an invalid token is parsed? Should we handle it as well?

Reply:
it will raise a 500 ISE exception. Not sure if we have to handle this (client will handle this correctly. It also shouldn't send requests like this in the first place).

This `except` is specifically to return `401`, because this will force a redirect to the login page for the web UI (and give the client a chance to reauthenticate)